lib/plugins/aws/package/compile/events/apiGateway/lib/deployment.js

```diff
@@ -13,7 +13,6 @@ module.exports = {
         Type: 'AWS::ApiGateway::Deployment',
         Properties: {
           RestApiId: { Ref: this.apiGatewayRestApiLogicalId },
-          StageName: this.options.stage,
```

Comment:
The stage is now created with Stage resource, it has the same stage name. I'm not sure if it makes any difference if the stage is created with the resource or automatically under the hood.